### ISSUE TITLE
Creating roles in cloud code

### ISSUE THREAD

**Issue by ball1008** [ParsePlatform/parse-server#853]
Make sure these boxes are checked before submitting your issue -- thanks for reporting issues back to Parse Server!

-[x ] You've met the [prerequisites](https://github.com/ParsePlatform/parse-server/wiki/Parse-Server-Guide#prerequisites).

-[x ] You're running the [latest version](https://github.com/ParsePlatform/parse-server/releases) of Parse Server.

-[x ] You've searched through [existing issues](https://github.com/ParsePlatform/parse-server/issues?utf8=%E2%9C%93&q=). Chances are that your issue has been reported or resolved before.

#### Environment Setup
Horoku button deployment

#### Steps to reproduce
I have deployed one of my apps to Horoku.  Thanks to help from Drew, I was able to deploy my cloud code.  Everything works prior to my first cloud code call which calls a function that adds a role. The only change I made was to comment out the old call to use master key and use the new useMasterKey:true.

Parse.Cloud.define("addToRole", function(request, response) {
                   //Parse.Cloud.useMasterKey();
                   var roleName = request.params.roleName;
                   var userId = request.params.user;
                   var user = request.user;
                   var queryRole = new Parse.Query("_Role"); // You need to get role object
                   queryRole.equalTo("name", roleName);
                   queryRole.first().then(function(role){
                                                             role.getUsers().add(user);
                                          return role.save({},{useMasterKey: true});
                                          
                                          }).then(function(){
                                                  response.success("success");
                                                  });
                   
                   });

#### Logs/Trace
Heroic logs show a timeout error
at=error code=H12 desc="Request timeout" method=POST path="/parse/functions/addToRole"

**Comment by flovilmart** [comment]
It seems that you don't handle errors.
Can you add a fail close to understand where it fails? Are you sure request.user is set?

**Comment by ball1008** [comment]
Your suggestion to add a fail close was perfect.  I found that the issue was that my cloud code was not connecting to the APIs.  I set the parse server url in my client app so it was connecting and running parse functions.  I did not have my config vars correct so cloud code could not connect.  Once I set the config var for the parse server correctly, I just had to add useMasterKey to a couple of other calls and everything is now working except my Stripe calls which I'll look into next.  Thanks for the help!
